gtsfm/evaluation/visualize_benchmark_comparison.py

```diff
@@ -0,0 +1,254 @@
+"""
+Use Plotly to create a heatmap representing a 2d table, and add annotations to it.
+
+Authors: John Lambert
+"""
+
+import argparse
+from collections import defaultdict
+from typing import List
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import plotly
+import plotly.graph_objects as go
+from matplotlib import colors
+from matplotlib.colors import LinearSegmentedColormap
+from plotly.graph_objs.layout import Font, Margin, XAxis, YAxis
+
+import gtsfm.evaluation.merge_reports as report_utils
+import gtsfm.evaluation.metrics_report as metrics_report
+
+
+HEATMAP_WIDTH = 1500
+HEATMAP_HEIGHT = 900
+NUM_COLORS_COLORMAP = 100
+EPSILON = 1e-12
+MAX_NUM_CHARS_ARTIFACT_FNAME = 35
+
+MIN_RENDERABLE_PERCENT_CHANGE = -20
+MAX_RENDERABLE_PERCENT_CHANGE = 20
+
+ZIP_FNAMES = [
+    "deep_front_end-2011205_rc3-20-png-wget-astronet-1024-true.zip",
```

Comment:
should this list come from some configuration file associated with the CI? Or just a directory read?

Reply:
thanks for the review Frank. We have a few options we could go with here. The zip artifact names are auto-generated, so we don't have them stored somewhere else yet. But we could put their names in a separate file somewhere.

If we glob the file names from a directory read, the directory might be missing some artifacts (if some of the CI runs failed, for example), so we might not now the dashboard is incomplete.

@akshay-krishnan @ayushbaid any preferences on your end here? I think it's ok for now the way it is, but if you have a strong preference, I'm happy to update

Reply:
If it's not too difficult, generating the file names from the information in `config_dataset_info` of the `benchmark.yaml` config would make the most sense to me. That way, we don't have to edit anything in the main code base when we add an additional benchmark.

Reply:
@johnwlambert Would this be possible to do?

Reply:
Sure, this is a good suggestion. I've updated the code now to parse the benchmark YAML and auto-generate the expected file names.